prototypes/entity.lua

```diff
@@ -1,302 +1,148 @@
+    icon_size = 32,
+		flags = {"placeable-neutral", "placeable-player", "player-creation"},
+		minable = {mining_time = 2, result = "warehouse-basic"},
+		max_health = 450,
+		corpse = "big-remnants",
+		dying_explosion = "medium-explosion",
+		open_sound = { filename = "__base__/sound/machine-open.ogg", volume = 0.85 },
+		close_sound = { filename = "__base__/sound/machine-close.ogg", volume = 0.75 },
+		vehicle_impact_sound = { filename = "__base__/sound/car-metal-impact.ogg", volume = 0.65 },
+		resistances =
+		{
+			{
+				type = "fire",
+				percent = 90
+			}
+		},
+		collision_box = {{-2.7, -2.7}, {2.7, 2.7}},
+		selection_box = {{-3.0, -3.0}, {3.0, 3.0}},
+		fast_replaceable_group = "container",
+		inventory_size = 800,
+		picture =
+		{
+			filename = "__Warehousing__/graphics/entity/warehouse-basic.png",
+			priority = "high",
+			width = 260,
+			height = 240,
+			shift = {1.0, -0.3},
+		},		
+		circuit_wire_max_distance = 7.5,
+		circuit_wire_connection_point =
+		{
+			shadow =
+			{
+				red = {2.01, 0.6},
+				green = {2.52, 0.6}
+			},
+			wire =
+			{
+				red = {1.71, 0.3},
+				green = {2.22, 0.3}
+			}
+		},
+	},
+})
+
+-- generate logistic variants
+function createLogisticContainer(name, logistic_type)
+  local p = table.deepcopy(data.raw["container"][name.."-basic"])
+  p.name = name.."-"..logistic_type
+  p.minable.result = p.name
+  p.icon = "__Warehousing__/graphics/icons/"..p.name..".png"
+  p.picture.filename = "__Warehousing__/graphics/entity/"..p.name..".png"
+  p.type = "logistic-container"
+  p.logistic_mode = logistic_type
+  return p
+end
+
+local storehouse_active_provider = createLogisticContainer("storehouse", "active-provider")
+local storehouse_passive_provider = createLogisticContainer("storehouse", "passive-provider")
+local storehouse_storage = createLogisticContainer("storehouse", "storage")
+storehouse_storage.inventory_size = 300
+local storehouse_buffer = createLogisticContainer("storehouse", "buffer")
+storehouse_buffer.logistic_slots_count = 12
+local storehouse_requester = createLogisticContainer("storehouse", "requester")
+storehouse_requester.logistic_slots_count = 12
+local warehouse_active_provider = createLogisticContainer("warehouse", "active-provider")
+local warehouse_passive_provider = createLogisticContainer("warehouse", "passive-provider")
+local warehouse_storage = createLogisticContainer("warehouse", "storage")
+warehouse_storage.inventory_size = 1000
+local warehouse_buffer = createLogisticContainer("warehouse", "buffer")
+warehouse_buffer.logistic_slots_count = 12
+local warehouse_requester = createLogisticContainer("warehouse", "requester")
+warehouse_requester.logistic_slots_count = 12
+
+
+data:extend({
+  storehouse_active_provider, 
+  storehouse_passive_provider,
+  storehouse_storage,
+  storehouse_buffer,
+  storehouse_requester,
+  warehouse_active_provider,
+  warehouse_passive_provider,
+  warehouse_storage,
+  warehouse_buffer,
+  warehouse_requester

```

Comment:
I'm not that well-versed in Lua, but I believe trailing commas are allowed. Using one here would be more future-proof against adding/removing warehouse types. (Just in terms of diff cleanliness, of course.)
```suggestion
  warehouse_requester,
```

Reply:
Lua does allow trailing comma on lists, I use them myself a fair bit whenever I'm uncertain if I need to add more.